### PR TITLE
[+] Implemented AlterEntryUseCase 

### PR DESCRIPTION
**Pull request by Yuhan-Ut**
[+]primary implements of alter entry usecase
todo: more discussion need for the alter_demo

### CODE REVIEW

**Comment by Yuhan-Ut** [comment]
@Yuhan-Ut
[[+]add simple description for entities' instance variables, for using convenience

**Review by Lei-Tin — COMMENTED** [review]
Overall, it seems like the `UseCase` is functioning pretty well!

 Just some issues: 
# Major issue
- We might not require a ViewModel and a Presenter at all in this UseCase, as the UseCase is only responsible for modifying the entry and reflecting the change in our database. So, we can look into removing it as it is unnecessary. The `BillUpdateUseCase` exists for the reason of updating the bill, so you do not have to call updateBill in this UseCase to ensure that the display works. 

# Minor issues
- I've been seeing formatting issues everywhere, e.g. "; //" and ";//". Make sure that if you decide to write comments or code somewhere, try to make it consistent, so that we could have a better time when we decide to change our convention and / or refactor the whole project. 
- There are some random line breaks inserted in files that is not related to this Pull Request's goal, e.g. [Here](https://github.com/CSC207-2022F-UofT/course-project-bill-gates/pull/35/files#diff-dfc890d7350f52212d160580307e922b7e11e8ad52d9e5f05295daf164634254R11). Maybe we can try to remove that, so that it doesn't show too much files being changed in this Pull Request.

File: src/main/java/billgates/alter_entry/AlterEntryUseCase.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+
+    @Override
+    public void alterEntry(int billId, QueryEntryData entry) {
+        gateway.modifyEntry(billId, entry);
+        List<Entry> result = this.gateway.getBillData(User.getInstance().getCurrentBillID()).getEntries()
+                .stream().map(d -> d.toEntryBuilder().build()).toList();
+        System.out.println(result); // TODO remove this
+        this.presenter.updateBill(new AlterEntryResponseModel(result.stream().map(Entry::toObjects).toList()));
```
[the remark this answers is not in the document]
Comment (Yuhan-Ut) [reply src/main/java/billgates/alter_entry/AlterEntryUseCase.java:26]: so, does this mean the presenter and output port is not necessary, we just need to call update usecase every time we use this Alter usecase?

File: src/main/java/billgates/entities/AbstractEntry.java [hunk 2]
```diff
@@ -7,14 +7,14 @@
...
-    private Attribute<ZonedDateTime> date;
-    private Attribute<Double> value;
-    private Attribute<String> currency;
-    private Attribute<String> description;
-    private Attribute<String> from;
-    private Attribute<String> to;
-    private Attribute<String> location;
+    private Attribute<Integer> id; //Entry id,
+    private Attribute<ZonedDateTime> date; // enteried date,
+    private Attribute<Double> value;// value of money
```
[the remark this answers is not in the document]
Comment (Yuhan-Ut) [reply src/main/java/billgates/entities/AbstractEntry.java]: fixed this, thanks to point out!

File: src/main/java/billgates/alter_entry/AlterEntryUseCase.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ScottCTD) [reply src/main/java/billgates/alter_entry/AlterEntryUseCase.java:26]: yes.

**Comment by Yuhan-Ut** [comment]
[https://github.com/CSC207-2022F-UofT/course-project-bill-gates/pull/41](url)
already make a light version of this usecase, and clean up commits inside the branch

**Pull request closed without merging**